### PR TITLE
MCS-358 - Add plausability parameters to step function

### PR DESCRIPTION
**Pull request by rartiss55**
See updated API description below for more detail on the new parameters. 

The following will be added to the scene history: choice, confidence, violations_xy_list, internal_state. 

If a heatmap_img is set in the step function, that will be uploaded to S3 (but the appropriate AWS config file values need to be set for this to work).

### CODE REVIEW

File: python_api/machine_common_sense/controller_ai2thor.py [hunk 1]
```diff
@@ -582,13 +612,56 @@ def step(self, action, **kwargs):
...
+            team_prefix = (self._config[self.CONFIG_TEAM] +
+                           '_') if self.CONFIG_TEAM in self._config else ''
+            file_name_prefix = team_prefix + self.HEATMAP_IMAGE_PREFIX + '_'
+
+            self.upload_image_to_s3(heatmap_img, file_name_prefix,
+                                    str(self.__step_number))
+
         return output
 
+    def upload_image_to_s3(self, image_to_upload: PIL.Image.Image,
```
Comment (rartiss55) [inline python_api/machine_common_sense/controller_ai2thor.py:636]: Ended up reusing what was already here for uploading object/depth masks to S3, but let me know if anything should be different for the heatmaps besides the file name.

File: python_api/machine_common_sense/util.py [hunk 2]
```diff
@@ -74,7 +74,9 @@ def generate_pretty_object_output(object_list):
             [
                 metadata.uuid,
                 metadata.shape,
-                ", ".join(metadata.texture_color_list),
+                ", ".join(
```
Comment (rartiss55) [inline python_api/machine_common_sense/util.py:77]: A couple unrelated to the ticket fixes in this file (this one I found when trying out the config file with metadata: "none")

File: python_api/API.md [hunk 3]
```diff
@@ -155,9 +155,10 @@ returns the scene output data object.
 
 
 
-#### step(action, \*\*kwargs)
+#### step(action: str, choice: str = None, confidence: float = None, violations_xy_list: List[Dict[str, float]] = None, heatmap_img: PIL.Image.Image = None, internal_state: object = None, \*\*kwargs)
```
Comment (rartiss55) [inline python_api/API.md:158]: Heads up - at some point when this autogenerated, it was referencing the Image.py path on my machine. Something to look out for.

File: python_api/machine_common_sense/controller_ai2thor.py (hunk 1 above)
Comment (deanwetherby) [reply python_api/machine_common_sense/controller_ai2thor.py:636]: @rartiss55 This is fine for now. I am planning to modify the upload for all of our image output to write/upload a single video file per image stream.

**Review by deanwetherby — APPROVED** [review]

**Review by brianpippin — APPROVED** [review]

**Pull request merged**